Build failure with mpfr-4.1.1 and cgal-5.5

OpenSCAD fails to build against mpfr-4.1.1 and cgal-5.5.
This is most likely an upstream bug in MPFR:
* https://github.com/CGAL/cgal/issues/7064
* https://gitlab.inria.fr/mpfr/mpfr/-/issues/1
* https://www.mpfr.org/mpfr-4.1.1/#bugs

Looks like it will be fixed in mpfr-4.2

Was:

Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin21.6.0
OS: macOS 12.6.1 Intel
openscad @2015.03-3_9 (installed using macport)

```
:info:build In file included from src/LibraryInfo.cc:15:
:info:build In file included from src/cgal.h:24:
:info:build In file included from /opt/local/include/CGAL/Gmpq.h:26:
:info:build In file included from /opt/local/include/CGAL/Gmpz.h:34:
:info:build In file included from /opt/local/include/CGAL/Gmp_coercion_traits.h:33:
:info:build /opt/local/include/CGAL/GMP/Gmpfr_type.h:208:25: error: cannot initialize a variable of type 'mpfr_ptr' (aka '__mpfr_struct *') with an lvalue of type 'mpfr_srcptr' (aka 'const __mpfr_struct *')
:info:build                         mpfr_custom_get_kind(f),
:info:build                         ^~~~~~~~~~~~~~~~~~~~~~~~
:info:build /opt/local/include/mpfr.h:1030:14: note: expanded from macro 'mpfr_custom_get_kind'
:info:build     mpfr_ptr _x = (x);                                                  \
:info:build              ^
:info:build /opt/local/include/mpfr.h:1010:9: note: expanded from macro 'mpfr_custom_init_set'
:info:build   _k = (k);                                                    \
:info:build         ^
```

--- a/scripts/macosx-build-dependencies.sh
+++ b/scripts/macosx-build-dependencies.sh
@@ -39,7 +39,7 @@ PACKAGES=(
     "boost 1.81.0"
     "eigen 3.4.0"
     "gmp 6.2.1"
-    "mpfr 4.1.0"
+    "mpfr 4.2.0"
     "glew 2.2.0"
     "gettext 0.21.1"
     "libffi REMOVE"

